components/tikv_util/src/codec/stream_event.rs

```diff
@@ -1,13 +1,56 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use crate::Either;
 use bytes::{Buf, Bytes};
 use std::io::prelude::*;
 use std::io::Cursor;
-use tikv_util::Either;
 
-pub struct Encoder;
+pub struct EventIterator {
+    buf: Cursor<Vec<u8>>,
+    index: usize,
+    len: usize,
+    pub val: Vec<u8>,
+}
+
+impl EventIterator {
+    pub fn new(buf: Vec<u8>) -> EventIterator {
+        let len = buf.len();
+        EventIterator {
+            buf: Cursor::new(buf),
+            index: 0,
+            len,
+            val: vec![],
+        }
+    }
+}
+
+impl Iterator for EventIterator {
+    type Item = Vec<u8>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.index >= self.len {
```

Comment:
Maybe we can check whether the `read` returns a `UnexpectedEOF` here for deciding whether we should stop?

Reply:
after change the trait to `fn next(&mut self) -> Result<()>`. I think this issue is fixed.